goinstant/docker_test.go

```diff
@@ -0,0 +1,948 @@
+			errorString:     nil,
+			testInfo:        "runCommand - run basic docker ps test",
+			execCommandMock: exec.Command,
+		},
+		{
+			commandName:     "docker",
+			suppressErrors:  nil,
+			commandSlice:    []string{"volume", "rm", "test-volume"},
+			pathToPackage:   "",
+			errorString:     fmt.Errorf("Error waiting for Cmd. Error: No such volume: test-volume\n: exit status 1"),
+			testInfo:        "runCommand - removing nonexistant volume should return error",
+			execCommandMock: exec.Command,
+		},
+		{
+			commandName:     "docker",
+			suppressErrors:  []string{"Error: No such volume: test-volume"},
+			commandSlice:    []string{"volume", "rm", "test-volume"},
+			pathToPackage:   "",
+			errorString:     nil,
+			testInfo:        "runCommand - error thrown should be suppressed",
+			execCommandMock: exec.Command,
+		},
+		{
+			commandName:    "git",
+			suppressErrors: nil,
+			commandSlice:   []string{"clone", "git@github.com:testhie/test.git"},
+			pathToPackage:  "test",
+			errorString:    nil,
+			testInfo:       "runCommand - clone a custom package and return its location",
+			execCommandMock: func(commandName string, commandSlice ...string) *exec.Cmd {
+				cmd := exec.Command("pwd")
+				return cmd
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.testInfo, func(t *testing.T) {
+			execCommand = tt.execCommandMock
+			pathToPackage, err := runCommand(tt.commandName, tt.suppressErrors, tt.commandSlice...)
+			if !assert.Equal(t, pathToPackage, tt.pathToPackage) {
+				t.Fatal("RunCommand failed - path to package returned is incorrect " + pathToPackage)
+			}
+			if err != nil && tt.errorString != nil && !assert.Equal(t, err.Error(), tt.errorString.Error()) {
+				t.Fatal("RunCommand failed - error returned incorrect")
+			}
+
+			if (err != nil && tt.errorString == nil) || (err == nil && tt.errorString != nil) {
+				log.Fatal("RunCommand failed - error returned incorrect")
+			}
+
+			t.Log(tt.testInfo + " passed!")
+		})
+	}
+}
+
+func Test_mountPackage(t *testing.T) {
+	defer gock.Off()
+	defer resetMountPackageMocks(runCommand, unzipPackage, untarPackage)
+
+	testCases := []struct {
+		pathToPackage    string
+		mockServer       func()
+		testInfo         string
+		wantErr          bool
+		errorString      string
+		runCommandMock   func(commandName string, suppressErrors []string, commandSlice ...string) (pathToPackage string, err error)
+		unzipPackageMock func(zipContent io.ReadCloser) (pathToPackage string, err error)
+		untarPackageMock func(tarContent io.ReadCloser) (pathToPackage string, err error)
+	}{
+		{
+			pathToPackage:  "http://test:8080/test",
+			mockServer:     func() {},
+			errorString:    "Error in downloading custom package",
+			wantErr:        true,
+			testInfo:       "mountPackage - should return error when downloading custom package fails",
+			runCommandMock: func(commandName string, suppressErrors []string, commandSlice ...string) (pathToPackage string, err error) {
```

Comment:
I think naming this function 'runCommandMock()' is misleading. Maybe 'mockRunCommand()' would be better?